corehq/messaging/scheduling/static/scheduling/js/conditional_alert_main.js

```diff
@@ -51,19 +51,26 @@ hqDefine(
 
             var initialFormContent = getFormContents();
 
-            $(".main-form").on("submit", function (event) {
+            $(".main-form").on("submit", function () {
                 var finalFormContent = getFormContents();
                 // compare changes with initialFormContent to check if only 'message' has changed
-                // to convey to HQ to reprocess or not
                 var formContentChanges = [];
                 Object.keys(finalFormContent).forEach(function (key) {
                     finalFormContent[key] !== initialFormContent[key]
                         ? formContentChanges.push(key)
                         : null;
                 });
-                if (_.isEqual(["message"], formContentChanges)) {
-                    // make the tweak
-                }
+                var runMessagingRule = !_.isEqual(
+                    ["message"],
+                    formContentChanges
+                );
+                // to convey to HQ to reprocess or not
+                var input = $("<input />")
+                    .attr("type", "hidden")
+                    .attr("name", "runMessagingRule")
+                    .attr("value", runMessagingRule.toString());
```

Comment:
nitpick: this will be a string by the time it gets to python even without the explicit cast, right?

Reply:
true, seems we don't need to explicitly call toString. Removing.